vissl/data/ssl_dataset.py

```diff
@@ -221,7 +227,25 @@ def _load_labels(self):
                 raise ValueError(f"unknown label source: {label_source}")
             self.label_objs.append(labels)
 
-    def __getitem__(self, idx):
+    def _init_subset(self):
+        if not self._subset_initialized:
+            if not self.data_limit_sampling.BALANCED:
+                self.sub_set = unbalanced_sub_sampling(
+                    total_size=len(self.data_objs[0]),
```

Comment:
For the moment, there are a few that seems to be made with the assumption that we have 1 data source, or that all data sources have the same length. I think this code does not make further assumption (*).

I am actually not quite sure where to push this in the future. The way I see it, we need to rework a bit the way we combine datasets. In particular, combining several datasets (of different length) means reworking the `ssl_dataset.py` module such that:
- combining means concatenating (which is different from the current behaviour, which, if we have 2 dataset, is such that we draw two samples, one for each dataset, at each call to __getitem__)
- labels of different kind can be concatenated (meaning that 2 datasets with different labels means twice as much labels, and optionally deal with the overlap?)

(*) In the current state of the code, the `_init_subset` function does not actually requires to have just one dataset: we can have several, as long as they have the same length, but the sampling will be performed on the first one (which means that the "balanced" constraint might not be respected for the second source, but overall the code will run fine).

What we can do is add an assert for the balanced in case we have either "no label" or "more than 1 data source".